llvm-hs/src/LLVM/Internal/ObjectFile.hs

```diff
@@ -0,0 +1,40 @@
+{-# LANGUAGE MultiParamTypeClasses #-}
+
+module LLVM.Internal.ObjectFile (
+    ObjectFile,
+    createObjectFile,
+    disposeObjectFile
+  ) where
+
+import Control.Monad.IO.Class
+import Control.Monad.AnyCont
+import Foreign.Ptr
+
+import LLVM.Prelude
+import LLVM.Internal.Coding
+import LLVM.Internal.MemoryBuffer
+import qualified LLVM.Internal.FFI.ObjectFile as FFI
+
+data ObjectFile = ObjectFile (Ptr FFI.ObjectFile)
+
+instance MonadIO m => EncodeM m ObjectFile (Ptr FFI.ObjectFile) where
```

Comment:
We mostly use explicit wrapping and unwrapping for things that are just newtypes rather than `EncodeM/DecodeM` instances so for consistency it would be nice to do this here as well.